coins/monero/src/wallet/mod.rs

```diff
@@ -218,4 +235,32 @@ impl Scanner {
       self.pair.view.deref() * spend,
     )
   }
+
+  /// Returns a featured address.
+  ///
+  /// A `Scanner` will fail to scan this address unless it is appropriate to the `Scanner`.
+  /// If the `Scanner` was defined as not handling the burning bug, by passing `None` for
+  /// `burning_bug` upon construction, this must be a guaranteed address
+  /// in order to be scanned by it.
+  pub fn featured_address(
+    &mut self,
+    subaddress: Option<(u32, u32)>,
+    payment_id: Option<[u8; 8]>,
+    guaranteed: bool,
+  ) -> MoneroAddress {
+    let is_subaddress = subaddress.is_some() && (subaddress.unwrap() != (0, 0));
+
+    let mut spend = self.pair.spend;
+    let mut view: EdwardsPoint = self.pair.view.deref() * &ED25519_BASEPOINT_TABLE;
+    if is_subaddress {
+      spend =
+        self.pair.spend + (&self.pair.subaddress(subaddress.unwrap()) * &ED25519_BASEPOINT_TABLE);
```

Comment:
Let's not have two instances of this cryptography/(0, 0) handling. We should have a `fn subaddress_derivation` function, handling calculation + registration, that both this and subaddress call.